sentry-core/src/main/java/io/sentry/core/PiiEventProcessor.java

```diff
@@ -0,0 +1,33 @@
+package io.sentry.core;
+
+import io.sentry.core.protocol.User;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Removes personal identifiable information from {@link SentryEvent} if {@link
+ * SentryOptions#isSendDefaultPii()} is set to false.
+ */
+@ApiStatus.Internal
+final class PiiEventProcessor implements EventProcessor {
+  private final @NotNull SentryOptions options;
+
+  PiiEventProcessor(final @NotNull SentryOptions options) {
+    this.options = options;
+  }
+
+  @Override
+  public SentryEvent process(SentryEvent event, @Nullable Object hint) {
+    if (!options.isSendDefaultPii()) {
+      final User user = event.getUser();
```

Comment:
do we have any PII set (like IP) in `event.request` that we should strip off as well? maybe in `event.request.headers` if it has a fixed key.

Reply:
Good point. I also realised that we need to make sure that `PiiEventProcessor` is the last processor that runs, otherwise other processors added later can actually add PII.

Reply:
good point, I noticed that too, that should be the last one, but I dont see how we can do that easily, any other processor added by the client manually will be added later, we can guarantee that its the last one in the SDK, but not the last one overall, I guess.
ideas are welcome of course.

Reply:
I extracted `PiiEventProcessor` to separate field. Another option would be to have special type of list with method `addBefore(processor, Class<T>)` but I think in this case it would be overengineering.

Reply:
yeah liked the idea, I'm fine with it, maybe @bruno-garcia has opinions.

Reply:
I'd prefer to avoid adding any pii to begin with.